src/app/entity/entity-mapper.service.ts

```diff
@@ -48,6 +48,29 @@ export class EntityMapperService {
     );
   }
 
+  /**
+   * Loads a list of Entity from the database whose IDs contain the prefix of the given resultEntity class.
+   *
+   * @param entityType the entity's class type (this would just be <code>Entity</code> for a pure entity).
+   * @returns A Promise containing an array with the entities.
+   */
+  public loadAll<T extends Entity>(entityType: { new(id: string): T; }): Promise<T[]> {
+    let newEntity = new entityType('');
+    return this._db.getAll(newEntity.getPrefix()).then(
```

Comment:
Unfortunately not really. I've tried static methods, but those kind of suck in typescript, because they don't exist on instances. We need the class to make a call which we won't have whenever we work with generics.